generator/generator.go

```diff
@@ -1107,34 +1106,74 @@ func getAttributeScripts(attr parser.Attribute) (scripts []string) {
 		}
 	}
 	if attr, ok := attr.(*parser.ExpressionAttribute); ok {
-		name := html.EscapeString(attr.Name)
+		name := html.EscapeString(attr.Key.String())
 		if isScriptAttribute(name) {
 			scripts = append(scripts, attr.Expression.Value)
 		}
 	}
 	return scripts
 }
 
-func (g *generator) writeBoolConstantAttribute(indentLevel int, attr *parser.BoolConstantAttribute) (err error) {
-	name := html.EscapeString(attr.Name)
-	if _, err = g.w.WriteStringLiteral(indentLevel, fmt.Sprintf(` %s`, name)); err != nil {
-		return err
+func (g *generator) writeAttributeKey(indentLevel int, attr parser.AttributeKey) (err error) {
+	if attr, ok := attr.(parser.ConstantAttributeKey); ok {
+		name := html.EscapeString(attr.Name)
+		if _, err = g.w.WriteStringLiteral(indentLevel, fmt.Sprintf(` %s`, name)); err != nil {
+			return err
+		}
+		return nil
+	}
+	if attr, ok := attr.(parser.ExpressionAttributeKey); ok {
+		var r parser.Range
+		vn := g.createVariableName()
+		// var vn string
+		if _, err = g.w.WriteIndent(indentLevel, "var "+vn+" string\n"); err != nil {
+			return err
+		}
+		// vn, templ_7745c5c3_Err = templ.JoinStringErrs(
+		if _, err = g.w.WriteIndent(indentLevel, vn+", templ_7745c5c3_Err = templ.JoinStringErrs("); err != nil {
+			return err
+		}
+		// p.Name()
+		if r, err = g.w.Write(attr.Expression.Value); err != nil {
+			return err
+		}
+		g.sourceMap.Add(attr.Expression, r)
+		// )
+		if _, err = g.w.Write(")\n"); err != nil {
+			return err
+		}
+		// Attribute expression error handler.
+		err = g.writeExpressionErrorHandler(indentLevel, attr.Expression)
+		if err != nil {
+			return err
+		}
+
+		// _, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(vn)
+		if _, err = g.w.WriteIndent(indentLevel, "_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString(` `+"+vn+"))\n"); err != nil {
+			return err
+		}
+		return g.writeErrorHandler(indentLevel)
 	}
 	return nil
```

Comment:
TODO: Add some sort of default case in case we add a new type later, and forget to update the generator for some reason.